src/app/shares/viewshareaccount/viewshareaccount.component.ts

```diff
@@ -0,0 +1,354 @@
+            waived: charge.currency.displaySymbol + charge.amountWaived,
+            outstanding: charge.currency.displaySymbol + charge.amountOutstanding,
+          };
+          this.chargeTableDataSource.push(chargeData);
+        }
+
+        this.chargeTableShow = true;
+      } else {
+        this.chargeTableShow = false;
+      }
+      if (data.status.value === 'Submitted and pending approval') {
+        this.buttons = {
+          singlebuttons: [
+            {
+              name: 'Modify Application',
+              icon: 'fa fa-pencil '
+            },
+            {
+              name: 'Approve',
+              icon: 'fa fa-check'
+            }
+          ],
+          options: [
+            {
+              name: 'Reject'
+            },
+            {
+              name: 'Delete'
+            }
+          ]
+        };
+      }
+
+      if (data.status.value === 'Approved') {
+        this.buttons = {
+          singlebuttons: [
+            {
+              name: 'Undo Approval',
+              icon: 'fa fa-undo'
+            },
+            {
+              name: 'Activate',
+              icon: 'fa fa-check'
+            }
+          ]
+        };
+      }
+
+      if (data.status.value === 'Active') {
+        if (this.sharesPendingForApproval) {
+          this.buttons = {
+            singlebuttons: [
+              {
+                name: 'Apply Additional Shares',
+                icon: 'fa fa-arrow-right'
+              },
+              {
+                name: 'Approve Additional Shares',
+                icon: 'fa fa-arrow-right'
+              },
+              {
+                name: 'Reject Additional Shares',
+                icon: 'fa fa-arrow-right'
+              },
+              {
+                name: 'Redeem Shares',
+                icon: 'fa fa-arrow-left'
+              }
+            ],
+            options: [
+              {
+                name: 'Close'
+              }
+            ]
+          };
+        } else {
+          this.buttons = {
+            singlebuttons: [
+              {
+                name: 'Apply Additional Shares',
+                icon: 'fa fa-arrow-right'
+              },
+              {
+                name: 'Redeem Shares',
+                icon: 'fa fa-arrow-left'
+              }
+            ],
+            options: [
+              {
+                name: 'Close'
+              }
+            ]
+          };
+        }
+
+        if (data.charges && data.charges.length !== 0) {
+          for (let i = 0; i < this.charges.length; i++) {
+            if (this.charges[i].name === 'Annual fee - INR') {
+              this.buttons.options.push({
+                name: 'applyAnnualFees'
+              });
+              this.annualChargeId = this.charges[i].id;
+            }
+          }
+        }
+      }
+    });
+  }
+
+  /**
+   * Gets Share Account Id from url params.
+   */
+  ngOnInit() {
+    this.route.params.subscribe(
```

Comment:
No need of a subscription. Use param map instead if this variable is being used.